BCnEnc.Net/Shared/DdsFile.cs

```diff
@@ -34,60 +34,36 @@ public static DdsFile Load(Stream s)
 					throw new FormatException("The file does not contain a dds file.");
 				}
 				var header = br.ReadStruct<DdsHeader>();
-				DdsHeaderDxt10 dxt10Header = default;
+				DdsHeaderDx10 dx10Header = default;
 				if (header.dwSize != 124)
 				{
 					throw new FormatException("The file header contains invalid dwSize.");
 				}
 
-				var dxt10Format = header.ddsPixelFormat.IsDxt10Format;
+				var dx10Format = header.ddsPixelFormat.IsDxt10Format;
 
 				DdsFile output;
 
-				if (dxt10Format)
+				if (dx10Format)
 				{
-					dxt10Header = br.ReadStruct<DdsHeaderDxt10>();
-					output = new DdsFile(header, dxt10Header);
+					dx10Header = br.ReadStruct<DdsHeaderDx10>();
+					output = new DdsFile(header, dx10Header);
 				}
 				else
 				{
 					output = new DdsFile(header);
 				}
 
 				var mipMapCount = (header.dwCaps & HeaderCaps.DdscapsMipmap) != 0 ? header.dwMipMapCount : 1;
-				var faceCount = (header.dwCaps2 & HeaderCaps2.Ddscaps2Cubemap) != 0 ? (uint)6 : (uint)1;
+				var faceCount = (header.dwCaps2 & HeaderCaps2.Ddscaps2Cubemap) != 0 ? 6u : 1u;
 				var width = header.dwWidth;
 				var height = header.dwHeight;
 
 				for (var face = 0; face < faceCount; face++)
 				{
-					var sizeInBytes = Math.Max(1, (width + 3) / 4) * Math.Max(1, (height + 3) / 4);
-					if (!dxt10Format)
-					{
-						if (header.ddsPixelFormat.IsDxt1To5CompressedFormat)
-						{
-							if (header.ddsPixelFormat.DxgiFormat == DxgiFormat.DxgiFormatBc1Unorm)
-							{
-								sizeInBytes *= 8;
-							}
-							else
-							{
-								sizeInBytes *= 16;
-							}
-						}
-						else
-						{
-							sizeInBytes = header.dwPitchOrLinearSize * height;
-						}
-					}
-					else if (dxt10Header.dxgiFormat.IsCompressedFormat())
-					{
-						sizeInBytes = (uint)(sizeInBytes * dxt10Header.dxgiFormat.GetByteSize());
-					}
-					else
-					{
-						sizeInBytes = header.dwPitchOrLinearSize * height;
-					}
+					var format = dx10Format ? dx10Header.dxgiFormat : header.ddsPixelFormat.DxgiFormat;
+					var sizeInBytes = GetSizeInBytes(format, width, height);
+
 					output.Faces.Add(new DdsFace(width, height, sizeInBytes, (int)mipMapCount));
 
 					for (var mip = 0; mip < mipMapCount; mip++)
```

Comment:
Can probably use right shift to divide the mip width and height below.

Reply:
True, doing a right shift by 1 should do the trick.

Reply:
Or by mip. `>> mip`